### ISSUE TITLE
AttributeError: 'list' object has no attribute 'iteritems' Ubuntu 14.04

### ISSUE THREAD

**Issue by qrrock** [metacloud/molecule#564]
Ubuntu 14.04
trying: molecule test
1. i got the same problem like https://github.com/metacloud/molecule/issues/548
2. cloned repo and checkouted to 548 them setup.py install
   no change
3. removed all images with <none> tag
   got:

Traceback (most recent call last):
  File "/usr/local/bin/molecule", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/molecule/cli.py", line 41, in main
    cli(obj={})
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 716, in **call**
    return self.main(_args, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 534, in invoke
    return callback(_args, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), _args, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/molecule/command/test.py", line 96, in test
    util.sysexit(t.execute()[0])
  File "/usr/local/lib/python2.7/dist-packages/molecule/command/test.py", line 46, in execute
    status, output = c.execute(exit=False)
  File "/usr/local/lib/python2.7/dist-packages/molecule/command/create.py", line 44, in execute
    self.molecule.driver.up(no_provision=True)
  File "/usr/local/lib/python2.7/dist-packages/molecule/driver/dockerdriver.py", line 145, in up
    cap_drop=container['cap_drop'])
  File "/usr/local/lib/python2.7/dist-packages/docker/api/container.py", line 157, in create_host_config
    return utils.create_host_config(_args, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/docker/utils/utils.py", line 752, in create_host_config
    host_config['PortBindings'] = convert_port_bindings(port_bindings)
  File "/usr/local/lib/python2.7/dist-packages/docker/utils/utils.py", line 285, in convert_port_bindings
    for k, v in six.iteritems(port_bindings):
  File "/usr/local/lib/python2.7/dist-packages/six.py", line 599, in iteritems
    return d.iteritems(**kw)
AttributeError: 'list' object has no attribute 'iteritems'

**Comment by retr0h** [comment]
This error is inside docker-py not molecule.  Have you made sure you are running the latest version of docker and docker-py?

**Comment by qrrock** [comment]
upgraded docker-engine to last release; no change

% molecule --version                                                     ~
molecule, version 1.12.6
% docker -v                                                              ~
Docker version 1.12.2, build bb80604
% pip show docker-py                                                     ~
---
Metadata-Version: 2.0
Name: docker-py
Version: 1.10.4
Summary: Python client for Docker.
Home-page: https://github.com/docker/docker-py/
Author: Joffrey F
Author-email: joffrey@docker.com
Installer: pip
License: UNKNOWN
Location: /usr/local/lib/python2.7/dist-packages
Requires: backports.ssl-match-hostname, ipaddress, docker-pycreds, requests, websocket-client, six
Classifiers:
  Development Status :: 4 - Beta
  Environment :: Other Environment
  Intended Audience :: Developers
  Operating System :: OS Independent
  Programming Language :: Python
  Programming Language :: Python :: 2
  Programming Language :: Python :: 2.6
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3
  Programming Language :: Python :: 3.3
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Topic :: Utilities
  License :: OSI Approved :: Apache Software License

**Comment by qrrock** [comment]
i have reposted this also to here; https://github.com/docker/docker-py/issues/1270 ;may by you can suggest something;

**Comment by retr0h** [comment]
Can I see your `molecule.yml`?  Is it possible to upload your role somewhere?

**Comment by shin-** [comment]
`port_bindings` in `create_host_config` needs to be a `dict`, but the project is passing a `list`. We could add a guard that would raise a more explicit exception on our end, but that doesn't change the fact that the type of the parameter is invalid.

[Reference](http://docker-py.readthedocs.io/en/stable/port-bindings/)

HTH!

**Comment by retr0h** [comment]
@qrrock have you sorted your issues?

**Comment by qrrock** [comment]
nope. just postponed for some time later, i think the end of the week. i've tried also other solutions. i'm going to refactor, and then get back to you if this is ok

**Comment by retr0h** [comment]
@qrrock can you paste me your molecule.yml file?

**Comment by retr0h** [comment]
@qrrock please send us an update.  Will be closing next week if we don't get further details. Thanks.

**Comment by qrrock** [comment]
I found code working on some other machine, so as I believe that must have been some kind of provisioning issue. so thanks for the help on this **bug_raport**. 

But, 

there is a **feature_request** though 

During the discussion we decided to put our master box ansible to work on container; Perhaps you could consider this option for molecule. 

exp. configuration check list
1. open ssh port on localhost machine;
2. configure localhost connection (key, etc), on ansible running on container (ssh)
3. switch molecule to create test containers on specified host in inventory (localhost)

**Comment by retr0h** [comment]
Closing since this bug report is a non-issue.  If you have feature requests please open a new issue and we will evaluate it.
